agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/detail/DetailScreen.kt

```diff
@@ -101,7 +113,39 @@ fun DetailScreen(
     state: DetailState,
     onAction: (DetailAction) -> Unit,
 ) {
+    val context = LocalContext.current
+    val itemModifier: Modifier = Modifier.padding(
+        horizontal = MaterialTheme.spacing.spaceMedium
+    )
+
+    val compressedPhotos = remember { mutableStateListOf<ByteArray?>() }
+    val compressedPhotosTemp = remember { mutableStateListOf<ByteArray?>() }
+
+    if (state.agendaType == AgendaType.EVENT) {
+        LaunchedEffect(state.details.asEventDetails.photos) {
+            compressedPhotosTemp.clear()
+            state.details.asEventDetails.photos.forEach { eventPhoto ->
+                val compressedByteArray = when(eventPhoto) {
+                    is EventPhoto.Local -> {
+                        withContext(Dispatchers.IO) {
+                            Uri.parse(eventPhoto.location).getCompressedByteArray(context)
+                        }
+                    }
+                    is EventPhoto.Remote -> {
+                        withContext(Dispatchers.IO) {
+                            URL(eventPhoto.location).getCompressedByteArray()
+                        }
+                    }
+                }
+                compressedPhotosTemp.add(compressedByteArray)
+            }
+            compressedPhotos.clear()
+            compressedPhotos.addAll(compressedPhotosTemp)
```

Comment:
How comes you decided to put this in the UI rather than the ViewModel? In the UI, the states will be reset after a config change (leading to re-compression each time) while in the ViewModel it wouldn't.

Also, with your current setup, every single photo is re-compressed if the photo list changes in any way (even if you delete a photo, the other existing photos will be compress again)